daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/ElementCombinator.scala

```diff
@@ -295,7 +295,7 @@ case class CaptureValueLengthStart(ctxt: ElementBase)
       new NadaParser(ctxt.erd)
 
   override lazy val unparser: Unparser =
-    if (ctxt.isReferencedByValueLengthUnparserExpressions) {
+    if (ctxt.isReferencedByValueLengthUnparserExpressions || ctxt.otherwiseShouldCaptureValueRegionLength) {
```

Comment:
Are there other reasons for "otherwiseShouldCapture..." than prefixLength? Maybe this should just be shouldCaptureForPrefixLength?.

Reply:
otherwiseShouldCapture wasn't added for prefix length. This was just moved out of the isReferencedByVvalueLength* stuff more consistent. It can be moved back if it makes sense.

Reply:
Just a suggestion that all the reasons for why one might lay down the "capture" things in the grammar should either be centralized (here), or distributed in some way. We are halfway between with one case (used in an expression) explicitly mentioned here, the others abstracted away in an "otherwise" method.

Reply:
Looking back, I think I moved things because I found the names confusing. The name isReferenedByVlueLengthUnparserExpressions should return wether or not something was refrenced by a value length expressions, and that's it. But if something wasn't referenced by a value length function, it could still return true if otherwiseSHouldCaptureValueLengthRegion was true. So the function could return something that seemed against what it said it would return. Perhaps the right fix is to just rename isReferencedByValueLengthUnparserExpression to shouldCaptureValueLengthRegionForUnparser or something? Then it does make more sense to have the otherwise stuff in there as well?

Reply:
Fixed with latest commit.